BattleTechModLoaderInjector/BTMLInjector.cs

```diff
@@ -103,11 +103,18 @@ private static int Main(string[] args)
                     if (injected) {
                         var yes = PromptForUpdateYesNo(requireKeyPress);
                         if (yes) {
-                            Restore(gameDllPath, gameDllBackupPath);
-                            // TODO: re-detect on backup and throw exception with useful
-                            //       mitigation steps if the backup is also fucked.
+                            if (File.Exists(gameDllBackupPath)) {
+                                Restore(gameDllPath, gameDllBackupPath);
+                            }
+                            else
+                            {
+                                SayException(new FileNotFoundException());
+                                SayHowToRecover();
```

Comment:
At this point where we cannot viably update, we need to bail on the process. I would say this is a good case for a custom exception. Something like `BackupFileNotFound`. If we throw that here, we can catch is specifically at the bottom and exit. That means we can move the check for it and the throw to the Restore function and keep the high level logic in this function easy to read and cyclomatic complexity lower.

Reply:
My simple thoughts were 'create the handler' and then something like

```cs
public class BackupFileNotFound: Exception {
    public BackupFileNotFound(string message): base(message) {
    }
}
```

This is the handler I basically copied from a beginner's guide.  I placed it in BTMLInjector.cs fwiw
```cs
if (!File.Exists(gameDllBackupPath))
{
    throw new BackupFileNotFound(gameDllBackupPath);
}
Restore(gameDllPath, gameDllBackupPath);
```

as I don't totally understand the exception protection you have setup I haven't thought ahead to how it will be handled.  More is needed I suspect but not sure what.  Also cyclomatic new word of the day, and useful!

or it seems to also make sense to put this into Restore() perhaps?
```csharp
private static void Restore(string filePath, string backupFilePath)
{
    if (!File.Exists(backupFilePath))
    {
        throw new BackupFileNotFound(backupFilePath + " was not found");
    }
    File.Copy(backupFilePath, filePath, true);
    WriteLine($"{Path.GetFileName(backupFilePath)} restored to {Path.GetFileName(filePath)}");
}
```

Is this on the right track?  thanks!

Reply:
I don't think you need the custom constructor on the exception class you defined. It should call the base constructor automatically AFAIK.

> or it seems to also make sense to put this into `Restore()` perhaps?

Yes, that is what I had advocated in my previous message. :)

Exception should get caught by [this block](https://github.com/janxious/BattleTechModLoader/blob/master/BattleTechModLoaderInjector/BTMLInjector.cs#L68-L72) and then exception message printed by it.